cairo/ethereum/cancun/vm/instructions/environment.cairo

```diff
@@ -0,0 +1,59 @@
+from starkware.cairo.common.bitwise import BitwiseBuiltin
+from starkware.cairo.common.math import split_felt
+
+from ethereum_types.bytes import Bytes32, Bytes32Struct
+from ethereum_types.numeric import U256, Uint, U256Struct
+
+from ethereum.cancun.vm import Evm, EvmImpl
+from ethereum.cancun.vm.exceptions import ExceptionalHalt
+from ethereum.cancun.vm.gas import charge_gas, GasConstants
+from ethereum.cancun.vm.stack import pop, push
+
+from ethereum.utils.numeric import U256_lt, U256_from_be_bytes
+
+// @notice Gets the versioned hash at a particular index
+func blob_hash{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, evm: Evm}() -> ExceptionalHalt* {
+    alloc_locals;
+
+    // STACK
+    let stack = evm.value.stack;
+    with stack {
+        let (index, err) = pop();
+        if (cast(err, felt) != 0) {
+            return err;
+        }
+    }
+
+    // GAS
+    let err = charge_gas(Uint(GasConstants.GAS_BLOBHASH_OPCODE));
+    if (cast(err, felt) != 0) {
+        return err;
+    }
+
+    // OPERATION
+    let blob_hashes = evm.value.env.value.blob_versioned_hashes;
+
+    // If index is within bounds, get the hash at that index
+    // Otherwise return zero bytes
+    let (high, low) = split_felt(blob_hashes.value.len);
+    let out_of_bounds = U256_lt(index, U256(new U256Struct(low, high)));
+    if (out_of_bounds.value == 0) {
+        tempvar blob_hash = Bytes32(new Bytes32Struct(0, 0));
+    } else {
+        tempvar blob_hash = blob_hashes.value.data[index.value.low];
```

Comment:
if we're only using index.value.low, we should probably assert index.value.high == 0, as well as blob_hashes.value.len.high ==0,
then just compare the two low felt values

Reply:
> we should probably assert index.value.high == 0,

no, this would be a deviation to spec. MAX_U256 would be a valid blob_hash index query and would return zero. We must not assume anything on the validity range of this input value.

for len(blob_hashes), we can assume that this number is way below 2**128.

so either:
1. just do u256 comparison, which is done now and is correct
2. return zero if input >= 2**128 or if input < len